test/built-ins/Array/prototype/at/coerced-index-resize.js

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.at
+description: >
+  Array.p.at behaves correctly on TypedArrays backed by resizable buffers when
+  the TypedArray is resized during parameter conversion
+includes: [resizableArrayBufferUtils.js]
+features: [resizable-arraybuffer]
+---*/
+
+function ArrayAtHelper(ta, index) {
+  const result = Array.prototype.at.call(ta, index);
+  return Convert(result);
+}
+
+function AtParameterConversionResizes(atHelper) {
+  for (let ctor of ctors) {
+    const rab = CreateResizableArrayBuffer(4 * ctor.BYTES_PER_ELEMENT, 8 * ctor.BYTES_PER_ELEMENT);
+    const fixedLength = new ctor(rab, 0, 4);
+    let evil = {
+      valueOf: () => {
+        rab.resize(2);
+        return 0;
+      }
+    };
+    assert.sameValue(atHelper(fixedLength, evil), undefined);
+  }
+  for (let ctor of ctors) {
+    const rab = CreateResizableArrayBuffer(4 * ctor.BYTES_PER_ELEMENT, 8 * ctor.BYTES_PER_ELEMENT);
+    const lengthTracking = new ctor(rab);
+    let evil = {
+      valueOf: () => {
+        rab.resize(2);
+        return -1;
+      }
+    };
+    // The TypedArray is *not* out of bounds since it's length-tracking.
+    assert.sameValue(atHelper(lengthTracking, evil), undefined);
+  }
+}
+
+AtParameterConversionResizes(ArrayAtHelper);
```

Comment:
I think I'd probably not structure these tests so that the test body takes place inside a function. That seemed more useful when they were all in one big file. (Same suggestion for the other test files.)

(The function name could be documentative, as well, but that role is already filled by the `description` in the front matter, which is more descriptive than the function name here anyway.)

Reply:
Done.